kong/runloop/handler.lua

```diff
@@ -1288,6 +1293,13 @@ return {
       var.upstream_x_forwarded_path   = forwarded_path
       var.upstream_x_forwarded_prefix = forwarded_prefix
 
+      do
+        local req_via = get_header(constants.HEADERS.VIA, ctx)
+        local kong_inbound_via = var.server_protocol .. " " .. SERVER_HEADER
```

Comment:
Probably better and faster to use this:
https://github.com/openresty/lua-nginx-module?tab=readme-ov-file#ngxreqhttp_version

Reply:
Done.